openapi/controller/main.py

```diff
@@ -0,0 +1,183 @@
+# -*- coding: utf-8 -*-
+import functools
+import hashlib
+import logging
+import os
+from ast import literal_eval
+import base64
+
+import odoo
+from odoo import http, SUPERUSER_ID, models
+from odoo.http import request, OpenERPSession
+from odoo.modules.registry import RegistryManager
+from psycopg2.extensions import ISOLATION_LEVEL_READ_COMMITTED
+
+from .pinguin import *
+
+_logger = logging.getLogger(__name__)
+
+#################################################################
+# Odoo REST API                                                 #
+#  Version 1                                                    #
+# --------------------------------------------------------------#
+# The current api version is considered stable, although        #
+# the exposed models and methods change as they are configured  #
+# on the database level. Only if significant changes in the api #
+# generation logic should be implemented in the future          #
+# a version bump should be considered.                          #
+#################################################################
+
+API_ENDPOINT = '/api'
+API_ENDPOINT_V1 = '/v1'
+# API_ENDPOINT_V2 = '/2'
```

Comment:
I think it's better to extract version from the url. Then we could add checking for the version and handle it appropriately::

    if version == 'v1':
         # support for previous version
         # ...
    # latest version
    # ...

Reply:
I can include that in the patched route :+1:

Reply:
I can include that in the patched route :+1: